### ISSUE TITLE
[BUG] Incorrect Use of Input Tokens Instead of Target Labels in Cross-Entropy Loss Calculation

### ISSUE THREAD

**Issue by oplex151** [homzer/MCC-KD#3]
Description:
There is an issue identified in the current implementation of the distill method. Specifically, when calculating the cross-entropy loss (CE Loss), the input tokens (example_a.tokens and example_b.tokens) are incorrectly used as target labels instead of the correct target labels (example_a.labels and example_b.labels). This causes the model to attempt to regenerate the input sequence rather than learning to predict the correct next token or output sequence.

at src/models/trainer.py \#368\#369

Code Snippet:
```
logits_a = self.model.forward(example_a.tokens).logits
logits_b = self.model.forward(example_b.tokens).logits
```

# Problematic part in the current implementation
```
ce_loss_a = self.criterion_ce.forward(logits_a, example_a.tokens)  # Incorrect: Should use labels, not tokens
ce_loss_b = self.criterion_ce.forward(logits_b, example_b.tokens)  # Incorrect: Should use labels, not tokens
```
Expected Behavior:
When calculating the cross-entropy loss, the model-generated logits should be compared against the target labels to ensure the model learns how to accurately predict the next token in the target sequence. The correct implementation should look like this:

# Correct approach
```
ce_loss_a = self.criterion_ce(logits_a.view(-1, logits_a.size(-1)), example_a.labels.view(-1))
ce_loss_b = self.criterion_ce(logits_b.view(-1, logits_b.size(-1)), example_b.labels.view(-1))
```
Impact:
Due to the current implementation's mistake, the model cannot effectively learn how to generate the correct output sequence, which may lead to degraded performance or training failure.

Suggested Fix:
Please modify the cross-entropy loss calculation part in the distill method to use example_a.labels and example_b.labels as target labels instead of example_a.tokens and example_b.tokens.
